projects/RabbitMQ.Client/client/impl/CommandAssembler.cs

```diff
@@ -121,6 +121,12 @@ private void ParseHeaderFrame(in InboundFrame frame)
                 throw new UnexpectedFrameException(frame.Type);
             }
 
+            if (totalBodyBytes == 0)
```

Comment:
There was an exception thrown when dispatching if the array was null (which was not caught due to an empty catch in the dispatcher, see other comment)